### PR TITLE
Align kubectl install and CI scripts to use local bin folder

### LINKED ISSUE #1192: Align install and CI scripts to use local bin folder for all tools in the repository
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
There are two different ways the repository installs tools that was found in #1097.  After discussion in office hours would like to align to install all tools via `makefile` into a `hack/tools/bin` and have all scripts use those tools.  

**What did you expect to happen:**
`make` and CI scripts all use the same version of tooling from local path.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]
see https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1097 for discussion

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by sayantani11**
**What type of PR is this?**

**What this PR does / why we need it**:
This PR aligns to install all tools via `makefile` into `hack/tools/bin` and have all tools used by  the scripts.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1192 

**Special notes for your reviewer**:
This PR isn't complete. It is up for review if the changes made till now are correct.

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by sayantani11** [comment]
@jsturtevant @CecileRobertMichon Review needed if changes made till now are correct.

File: Makefile [hunk 1]
```diff
@@ -242,8 +242,12 @@ $(ENVSUBST_BIN): $(ENVSUBST) ## Build envsubst from tools folder.
 $(GOLANGCI_LINT): ## Build golangci-lint from tools folder.
 	GOBIN=$(TOOLS_BIN_DIR) $(GO_INSTALL) github.com/golangci/golangci-lint/cmd/golangci-lint $(GOLANGCI_LINT_BIN) $(GOLANGCI_LINT_VER)
 
-$(KUSTOMIZE): ## Build kustomize from tools folder.
-	GOBIN=$(TOOLS_BIN_DIR) $(GO_INSTALL) sigs.k8s.io/kustomize/kustomize/v4 $(KUSTOMIZE_BIN) $(KUSTOMIZE_VER)
+$(KUSTOMIZE): ## Build kustomize
```
Comment (CecileRobertMichon) [inline Makefile:245]: I'm a little confused about what this PR is trying to achieve... I thought the point of #1671 was to make sure all scripts in the repo that use `kubectl` (and call ensure-kubectl.sh) use the makefile to install it and use the version in hack/tools/bin

I think a good example of that is how `apidiff` is installed and used in ci-apidiff.sh. We want the same thing for kubectl. @jsturtevant please correct me if I'm misunderstanding the issue.

Comment (sayantani11) [reply Makefile:245]: @CecileRobertMichon Guess even I am confused with the issue. I'll try to figure out.

Comment (CecileRobertMichon) [reply Makefile:245]: let me know if you need help, James is on vacation this week

Comment (sayantani11) [reply Makefile:245]: Sure @CecileRobertMichon but I guess it will need sometime

Comment (sayantani11) [reply Makefile:245]: @CecileRobertMichon Is there an example PR I could refer to? I am actually kind of lost with this.

**Pull request closed without merging**